internal/services/connection_check.go

```diff
@@ -125,7 +126,7 @@ func (cs *ConnectionService) connectionCheck() {
 	if cs.isDynamicScalingEnabled() || cs.canaryConfig.ExpectedClusterSize != len(cs.brokers) {
 		cs.brokers, _, err = cs.admin.DescribeCluster()
 		if err != nil {
-			if err.Error() == "EOF" {
+			if strings.Contains(err.Error(), util.ErrEof) || strings.Contains(err.Error(), util.ErrConnectionResetByPeer) {
```

Comment:
Is it intentional that you changed the condition from _is EOF_ to _contains EOF_?

Reply:
Just to generalize as the connection reset by peer that is longer.